### PR TITLE
Joyrostie/asset search issue#360

### LINKED ISSUE #360: AssetSearch show dropdown list in case of advanceSearch
Screen Shot 2019-06-25 at 12 29 16

also fix empty strings in metadata filter in case of empty fields in advance search form

Screen Shot 2019-06-25 at 12 26 38

### PR DESCRIPTION
**Pull request by joyrostie**
* Updated `AssetSearch` sdk query with empty Metadata object.
* Fixed `Search` showing live search results with advanced search prop.

Closes #360

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-365.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-365.gearboxjs.preview.cogniteapp.com

File: src/components/AssetSearch/AssetSearch.tsx [hunk 1]
```diff
@@ -70,10 +70,9 @@ export class AssetSearch extends React.Component<
       metadata:
         (query.advancedSearch &&
           query.advancedSearch.metadata &&
-          query.advancedSearch.metadata.reduce(
-            (a, c) => ({ ...a, [c.key]: c.value }),
-            {}
-          )) ||
+          query.advancedSearch.metadata
+            .filter(meta => meta.key !== '')
+            .reduce((a, c) => ({ ...a, [c.key]: c.value }), {})) ||
```
Comment (mrtraser) [inline src/components/AssetSearch/AssetSearch.tsx]: I would change it a bit. If you use `filter` and `reduce` you'll get `O(2n)` complexity. But you can avoid filtering by moving that logic to reduce callback (to acheave `O(n)` complexity):

```typescript
.reduce((a, c) => (c.key ? { ...a, [c.key]: c.value } : a), {}))
```
It's not so important in this case, but its better to think about complexity when you're formatting/modifining data

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-365.gearboxjs.preview.cogniteapp.com

File: src/components/AssetSearch/AssetSearch.tsx (hunk 1 above)
Comment (joyrostie) [reply src/components/AssetSearch/AssetSearch.tsx]: Complexity reductioned on both master and sdk-2.0 branches PRs.

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-365.gearboxjs.preview.cogniteapp.com

**Pull request merged**